osgpkitools/incommon_request.py

```diff
@@ -219,10 +220,12 @@ def submit_request(config, restclient, hostname, cert_csr, sans=None):
             logger.debug('response text: ' + str(response_text))
             response_data = json.loads(response_text)
             response_data = response_data['sslId']
-        else:
+        elif response.status == 401:
             raise AuthenticationFailureException(response.status, "Connection failure to InCommon API. Check your authentication credentials.")
+        else:
+            raise httplib.HTTPException(response.status, "Connection failure to InCommon API")
     except httplib.HTTPException as exc:
-        raise
+        raise httplib.HTTPException("Connection failure to InCommon API: " + exc) 
```

Comment:
I'm not sure if you can put an exception in a string, do you have any examples?  You could always do str(exc) to make sure.

Reply:
You're right. I'm printing the message first and then raise an httplib.HTTPException()